### ISSUE TITLE
PHP Fatal Error: Nesting level too deep - recursive dependency? (from CArrayDataProvider)

### ISSUE THREAD

**Issue by mattsawyer77** [yiisoft/yii#2082]
I'm running into an issue on a page that uses a CGridView backed by a CArrayDataProvider. From what I can tell there are only a handful of rows of data that should be loaded in this particular instance. The line that throws the error is in CArrayDataProvider::sortData() at

``` php
call_user_func_array('array_multisort', $args);
```

Also, other URLs that use this same page (i.e. different data returned) are working fine.

**Comment by fsateler** [comment]
I had the same issue. The problem was with a boolean field. I solved it by adding a second field to order by to the sort definition:

     'field' => [
        'asc' => ['field' => SORT_ASC, 'name' => SORT_ASC],
        'desc' => ['field' => SORT_DESC, 'name' => SORT_DESC],
    ],

I do not know why this happens

**Comment by fsateler** [comment]
Update: I can reproduce this with any field that has a repeat value.

**Comment by Bhoft** [comment]
Will this bug fixed? Because it would destroy the sort possibilities of every grid with an CArrayDataProvider if the same value would be in a column twice. Which is of course is happening for every boolean column but could also happen for other columns as well.
The solution couldn't be that we add the primary key to the sort options for each column where this error "could" happen.

**Comment by marcuzy** [comment]
Based on http://stackoverflow.com/questions/3834791/fatal-error-nesting-level-too-deep-recursive-dependency I resolve problem with follow code:
```php
        //$args[] = &$array;
        $arrayKeys = [];
        foreach ($array as $i => $item) {
            $arrayKeys[] = $i;
        }
        $args[] = &$arrayKeys;

        call_user_func_array('array_multisort', $args);

        $copyArray = $array;
        foreach ($arrayKeys as $i => $index) {
            $array[$i] = $copyArray[$index];
        }
```

**Comment by cebe** [comment]
@marcuzy are you on yii2? that should be fixed in 2.0.7 then.
